samples/QuickstartWeatherServer/Tools/HttpClientExt.cs

```diff
@@ -0,0 +1,14 @@
+
+using System.Text.Json;
+
+namespace ModelContextProtocol;
+
+internal static class HttpClientExtensions
+{
+    public static async Task<JsonDocument> ReadJsonDocument(this HttpClient client, string requestUri)
+    {
+        using var response = await client.GetAsync(requestUri);
+        response.EnsureSuccessStatusCode();
+        return await JsonDocument.ParseAsync(await response.Content.ReadAsStreamAsync());
```

Comment:
Note that this returns a `JsonDocument` that is backed by a pooled buffer so the result would need to be disposed eventually.

Reply:
Yup, I believe I added a `using` in all appropriate places.